### DISCUSSION TITLE
Add a configuration option to make any mob tamable, and define the item used to tame it

### DISCUSSION THREAD

**Discussion by The-Nyla** [PurpurMC/Purpur#443] [__Dismissed__] [answered]
Hello, I just wanted to suggest a useful configuration option that would allow you to enable/disable a value which could make any mob tameable, and allow you to modify the item used to tame it, it could also be used to alter the tameability, and items used to tame mobs that are tamable by default, like wolves, and foxes, you could also add an option to make them only tamable as a baby mob

**Comment by granny** [comment, marked as answer]
Excerpt from Billy
> // 03/26/2020
> i made cows tameable as an experiment
> worked well, but had odd side effects
> and cows are probably the simplest mobs in the game
> foxes have a lot of AI, so would be harder to make work without those side effects
> foxes are probably the most AI heavy mob in the game, iirc
> // 07/13/2020
> and it was quite the mess
> i have a video of that somewhere.. tameable cow that fights for you lol
> https://www.youtube.com/watch?v=YmZpbs_XQPg
> my fav part was when it tried to fight a zombie for me.. the zombie would hit it and the cow would panic, but try to fight at the same time
> ultimately, zombie killed the cow

TLDR; It's possible, but *very* difficult to accomplish. And that's coming from Billy, who made every mob in minecraft ridable. Hopefully it'll become a thing in the future, but I don't see it happening anytime soon.

**Reply by The-Nyla** [reply]
ok, some person actually made a plugin that made foxes tamable, but that's about as far as anyone has ever gotten, as nobody has done anything to allow any mob to be tamable

**Reply by granny** [reply]
> ok, some person actually made a plugin that made foxes tamable, but that's about as far as anyone has ever gotten, as nobody has done anything to allow any mob to be tamable

Could you share the plugin you're talking about?

**Reply by The-Nyla** [reply]
Sure!


https://www.spigotmc.org/resources/tamable-foxes-1-14-x-1-15-x-1-16-x-1-17-x.69537

**Reply by BillyGalbreath** [reply]
As mentioned, they did [a lot of hard work](https://github.com/SeanOMik/TamableFoxes/tree/master/1_17_1_R1/src/main/java/net/seanomik/tamablefoxes/versions/version_1_17_1_R1/pathfinding) for that. Remember, adding new behavior is hard to write, but easy to maintain. However, having to blend too much with old behavior is not only hard to write, but also hard to maintain each update. The maintenance burden alone would make such a patch get dropped on the first update that conflicts until someone wants to put in the work _again_ to update it.

If someone else wants to take on that, I wont stop them. ^_^ I just dont want that person doing the work to be me ;)

**Reply by granny** [reply]
I'd like to also mention that that's just for *one* mob. Take what that developer has and imagine what that would be like for *every mob in the game* (which would look similar to the [ridables patch](https://github.com/pl3xgaming/Purpur/blob/master/patches/server/0007-Ridables.patch) being over 5807 lines long).

**Reply by The-Nyla** [reply]
Yeah, I realized it would be difficult, now I see that that would be even more difficult than I thought, however I still think it would be a very useful addition to pupur, though it would probably be a very long-term project. For now I’ll just hope they add this feature in mythicmobs, as that would actually allow much more customization than purpur. 

Thank you!
